src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -654,6 +731,29 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 				calledMethodDefinition,
 				operation);
 
+			if (calledMethodDefinition.IsPInvokeImpl) {
+				// Is the PInvoke dangerous?
+				bool comDangerousMethod = IsComInterop (calledMethodDefinition.MethodReturnType, calledMethodDefinition.ReturnType);
+				foreach (ParameterDefinition pd in calledMethodDefinition.Parameters) {
+					TypeReference paramTypeReference = pd.ParameterType;
+					if (paramTypeReference is TypeSpecification) {
+						paramTypeReference = (paramTypeReference as TypeSpecification).ElementType;
+					}
+					TypeDefinition paramTypeDefinition = _context.TryResolve (paramTypeReference);
+					if (paramTypeDefinition != null) {
+						comDangerousMethod |= IsComInterop (pd, paramTypeReference);
+					}
+				}
+
+				if (comDangerousMethod) {
+					reflectionContext.AnalyzingPattern ();
+					reflectionContext.RecordUnrecognizedPattern (2050, $"P/invoke method '{calledMethodDefinition.GetDisplayName ()}' declares a parameter with COM marshalling. Correctness of COM interop cannot be guaranteed after trimming. Interfaces and interface members might be removed.");
+				}
+
+				// We did not handle this call
+				return false;
```

Comment:
This will prevent RUC on PInvoke method to work. I can imagine cases where that might be useful. Not counting that there's no reason for it to NOT work.

I would prefer if we move this check to the default branch in the switch.